src/skmultiflow/core/stream.py

```diff
@@ -0,0 +1,28 @@
+from streamz import Stream
+
+@Stream.register_api()
+class predict(Stream):
+    def __init__(self, upstream, model, **kwargs):
+        self.model = model
+        super().__init__(upstream, **kwargs)
+
+    def update(self, X, who=None):
+        return self._emit(self.model.predict(X))
+
+@Stream.register_api()
+class partial_fit(Stream):
+    def __init__(self, upstream, model, **kwargs):
+        self.model = model
+        super().__init__(upstream, **kwargs)
+
+    def update(self, x, who=None):
+        X, y = X
+        self.model.partial_fit(X, y)
+        return self._emit(self.model)
```

Comment:
Here is what I was talking about. I guess it's OK to emit the whole model (in-process, anyway), and then downstream streamz nodes can decide what to do with it to get out any metrics of interest. On the other hand, the model is mutable, so downstreams had better not be async, else they may process on incorrect state.

Is there any obvious "output" to the fitting and current state that we could consider as a result of the partial fit?

Reply:
Following the sklearn way of doing, the output of partial_fit should be `self.model`
But in this case we are in a different context, calls in streamz may be asynchronous, which get things complicated

Option 1 : 
Return self.model, and to make very explicit for users that this part of the code is not responsible for ensuring process on a correct state

Option 2 (requires more work):
Interally manage the access to self.model (via [tornado conditions ? ](https://www.tornadoweb.org/en/stable/locks.html))

Reply:
After some consideration, Option 1 is OK for the moment - with appropriate caveats - and I don't expect we actually expect any async down-streams here for the moment. In examples, it would be good to show what you might do with the model to get some basic metrics out for plotting the performance of the model versus time.

Reply:
@martindurant to answer to your question more deeply, most of the informations you could get from a model actually depend on the "state" of the model
Even something like a call to "get_infos()" would lead to incorrect results if runned asynchronously. So I don't see what useful information can be returned ...

Reply:
`.map(lambda x: x.get_infos())` would run immediately, generate a string, and be correct at the moment it is called. *If* there are downstream nodes that want to do something else with the model object, they may be slightly out-of-date, but not by much. If the object is to plot a time-series of the model performance, I don't anticipate much of a problem.

Reply:
I think I got your point
I would actually be OK for returning get_infos() for a V1

Reply:
`get_info` is just a method to provide the initial configuration of the estimator, so it does not change as the model is trained. I prefer `return self` because the user could then call something like `model.score(X, y_true)` to have an estimation on performance. `score()` is now available in all classifiers and regressors.